ticketutil/ticket.py

```diff
@@ -67,19 +105,21 @@ def _create_requests_session(self):
             r.raise_for_status()
             logging.debug("Create requests session: Status Code: {0}".format(r.status_code))
             logging.info("Successfully authenticated to {0}".format(self.ticketing_tool))
+            return s
         # We log an error if authentication was not successful, because rest of the HTTP requests will not succeed.
         # Instead of using e.message, use e.args[0] instead to prevent DeprecationWarning for exception.message.
         except requests.RequestException as e:
-            logging.error("Error authenticating to {0}. No valid kerberos principal found.".format(self.auth_url))
+            logging.error("Error authenticating to {0}.".format(self.auth_url))
             logging.error(e.args[0])
-        return s
+            s.close()
 
```

Comment:
Good point, @rpavelka :) I'm working on better exception handling and should be pushing an update in the next couple days. I'll open a PR for it and include you and @pzubaty.

So, for this specific case, I raise the exception in __init__() lines 45-47. If a session object is not returned from _create_requests_session(), then I raise a TicketException:

```
        self.s = self._create_requests_session()
        if not self.s:
            raise TicketException("Error authenticating to {0}.".format(self.auth_url))
```
I'm also adding some new checks in __init__() to verify the project and optional ticket_id being passed in actually exist. Because if those don't exist, a Ticket object should not be created and an exception should be raised.